src/ray/raylet/node_manager.h

```diff
@@ -454,10 +455,18 @@ class NodeManager : public rpc::NodeManagerServiceHandler {
   void HandleDisconnectedActor(const ActorID &actor_id, bool was_local,
                                bool intentional_disconnect);
 
+  /// Handle the case where a worker is available. This should be called when a new
+  /// worker is registered with a port greater than zero (which indicates it will
+  /// act as a rpc server), and when a worker finishes it assigned task, which is
+  /// indicated by a `GetTask` message (or a rpc reply if the worker is a rpc server).
+  ///
+  /// \param worker Worker that is available.
+  /// \return void.
```

Comment:
we should document clearly that pull mode is temporarily used for old workers, and eventually we'll only have push mode.

Reply:
makes sense. thanks.